codegen/generator/src/Visitors/PaginationVisitor.cs

```diff
@@ -0,0 +1,267 @@
+                                    foreach (var param in newInstance.Parameters)
+                                    {
+                                        if (param is VariableExpression varExpr && options.ParamsToReplace.Contains(varExpr.Declaration.RequestedName))
+                                        {
+                                            // Replace the parameter with the options parameter.
+                                            if (_paramReplacementMap.TryGetValue(varExpr.Declaration.RequestedName, out var replacement))
+                                            {
+                                                newParameters.Add(optionsParam.NullConditional().Property(replacement));
+                                            }
+                                        }
+                                        else if (param is InvokeMethodExpression invokeMethod && invokeMethod.MethodName == "ToString" &&
+                                                 invokeMethod.InstanceReference is NullConditionalExpression nullConditional &&
+                                                 nullConditional.Inner is VariableExpression varExpr2 &&
+                                                 options.ParamsToReplace.Contains(varExpr2.Declaration.RequestedName))
+                                        {
+                                            // Replace the parameter with the options parameter.
+                                            if (_paramReplacementMap.TryGetValue(varExpr2.Declaration.RequestedName, out var replacement))
+                                            {
+                                                newParameters.Add(optionsParam.NullConditional().Property(replacement).NullConditional().Invoke("ToString", Array.Empty<ValueExpression>()));
+                                            }
+                                        }
+                                        else
+                                        {
+                                            // Keep the original parameter.
+                                            newParameters.Add(param);
+                                        }
+                                    }
+                                    // Create a new ExpressionStatement with the same children as the original, but with the new parameters.
+                                    var newNewInstanceExpression = new NewInstanceExpression(
+                                        newInstance.Type,
+                                        newParameters);
+
+                                    var newKeywordExpression = new KeywordExpression(
+                                        keyword.Keyword,
+                                        newNewInstanceExpression);
+
+                                    var newExpressionStatement = new ExpressionStatement(newKeywordExpression);
+
+                                    return newExpressionStatement;
```

Comment:
Since we know they keyword expression is "return" could we simplify this to ?
```suggestion
                                    return Snippet.Return(Snippet.New(newInstance.Type, newParameters));
```

Reply:
It looks like Snippet.New is not available as a method (Non-invocable member 'Snippet.New' cannot be used like a method.)

Reply:
sorry, that's my fault. Can you try `Snippet.New.Instance` ?